File: test/device.py
from __future__ import absolute_import

from unittest import TestCase

from model_mf2c.classes import Device

from .user import test_instance as owner
from .static_info import test_instance as static_info
from .dynamic_info import test_instance as dynamic_info
from .network_info import test_instance as network_info
from .behaviour_info import test_instance as behaviour_info
from .security_info import test_instance as security_info
from .component import test_instance as component
from .sharing_model_info import test_instance as sharing_model_info

test_instance = Device(device_id="id", owner=owner, static_info=static_info, dynamic_info=dynamic_info, network_info=network_info, behaviour_info=behaviour_info, security_info=security_info, attached_components=[component, component], sharing_model_info=sharing_model_info)

class TestDevice(TestCase):
    def setUp(self):
        self.device = test_instance

    def test_init(self):
        self.assertEqual('id', self.device.device_id)
        self.assertIs(static_info, self.device.static_info)
        self.assertIs(dynamic_info, self.device.dynamic_info)
        self.assertIs(network_info, self.device.network_info)
        self.assertIs(behaviour_info, self.device.behaviour_info)
        self.assertIs( security_info, self.device.security_info)
        self.assertListEqual([component, component], self.device.attached_components)
        self.assertIs(sharing_model_info, self.device.sharing_model_info)

File: demos/cimi/src/model/Device.java
package model;

@SuppressWarnings("unused")
public class Device extends CIMIResource {
    // private User owner;
    private String id; // mF2C id of the device
    // StaticInfo
    private String operatingSystem;
    private String systemArchitecture;
    private String processorMaker;
    private String processorArchitecture;
    private int numCPUs;
    private String CPUClockSpeed;
    private int numCores;
    private float RAMSize;
    private float storageSize;
    private String agentType;
    private String actuatorInfo;
    private boolean limitedPower;
    private String graphicsCardInfo;
    private String CPU_info; // For the Landscaper collector
    private String hwloc; // For the Landscaper collector
    // private List<Component> attachedComponents;
    // private SharingModel sharingModel;
    // DynamicInfo
    // ... TBD
    // NetworkInfo
    // ... TBD
    // BehaviourInfo
    // ... TBD
    // SecurityInfo
    // ...TBD

    public Device(final String deviceId, final String deviceOS, final String deviceSysArch,
	    final String deviceProcMaker, final String deviceProcArch, final int deviceNumCPUs,
	    final String deviceClockSpeed, final int deviceNumCores, final float deviceRAMSize, final float deviceStorageSize, final String deviceAgentType, final String deviceActuatorInfo,
	    final boolean deviceLimitedPower, final String deviceGraphicsCard, final String resourceID,
	    final String resourceName, final String resourceDescription, String resourceURI) {

	// CIMI default fields
	super(resourceID, resourceName, resourceDescription, resourceURI);
	// Device fields
	this.id = deviceId;
	this.operatingSystem = deviceOS;
	this.systemArchitecture = deviceSysArch;
	this.processorMaker = deviceProcMaker;
	this.processorArchitecture = deviceProcArch;
	this.numCPUs = deviceNumCPUs;
	this.numCores = deviceNumCores;
	this.RAMSize = deviceRAMSize;
	this.storageSize = deviceStorageSize;
	this.agentType = deviceAgentType;
	this.actuatorInfo = deviceActuatorInfo;
	this.limitedPower = deviceLimitedPower;
	this.graphicsCardInfo = deviceGraphicsCard;
	this.CPU_info = null;
	this.hwloc = null;
    }

    public String getId() {
	return this.id;
    }

    public void setOperatingSystem(String newOperatingSystem) {
	this.operatingSystem = newOperatingSystem;
	resourceUpdated();
    }

    public void setNumCores(int newNumCores) {
	this.numCores = newNumCores;
	resourceUpdated();
    }

    public void setRAMSize(float newRAMSize) {
	RAMSize = newRAMSize;
	resourceUpdated();
    }

    public void setStorageSize(float newStorageSize) {
	this.storageSize = newStorageSize;
	resourceUpdated();
    }

    public void setCPUInfo(String cpuInfo) {
	this.CPU_info = cpuInfo;
	resourceUpdated();
    }

    public String getCPUInfo() {
	return this.CPU_info;
    }

    public void setHwloc(String hwloc) {
	this.hwloc = hwloc;
	resourceUpdated();
    }

    public String getHwloc() {
	return this.hwloc;
    }
}